packages/@lwc/engine-core/src/framework/vnodes.ts

```diff
@@ -104,7 +104,7 @@ export interface VNodeData {
     // All props are readonly because VElementData may be shared across VNodes
     // due to hoisting optimizations
     readonly props?: Readonly<Record<string, any>>;
-    readonly attrs?: Readonly<Record<string, string | number | boolean>>;
+    readonly attrs?: Readonly<Record<string, string | number | boolean | null | undefined>>;
```

Comment:
Not sure why we didn't have this before, because in practice this can definitely be `null` or `undefined`.